### ISSUE TITLE
Dlib's Chinese whispers clustering is not correctly clustering faces

### ISSUE THREAD

**Issue by s0r2637** [davisking/dlib#1454]
## Expected Behavior
I am using dlib.chinese_whispers_clustering.  The code was run on a list of face embeddings performed on a large subset of  LFW data.  The labels created by the clustering is not accurate.  In fact, it is pretty abysmal.  I am not doubting the dlib's face recognition, but only clustering algorithm.  The chinese whispers algorithm for a threshold tolerance of 0.56 claims that two embeddings are of the same person even though the Eucledian distance between them is larger than 0.56.  How confident are you about the Chinese whispers clustering algorithm or your implementation of it?

* **Version**: <19.15.0>
* **Where did you get dlib**: <pip install dlib inside Anaconda -->
* **Platform**: <No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial>
* **Compiler**: <gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.10)>

**Comment by davisking** [comment]
I think it's fine. Lots of people use it and it works great.  For instance, the example program that runs it and does face clustering works perfectly.  I've also used it myself a lot and it generally gives good clusters. However, it's obviously not going to give a perfect identity clustering in every case though.

**Comment by s0r2637** [comment]
Do you mind, if I send you a pandas dataframe pickle that contains the list
of embeddings and you can see what I am experiencing?  Also, do we need to
run your computer face descriptor in conjunction with your mmod or hog face
detector? I used OpenCV resnet face detector and passed that to your
compute face descriptor method.  The latter question is crucial to me.

Regards,

**Comment by s0r2637** [comment]
When you say as long as you get the same descriptor back, what are you
comparing with?  Are you suggesting to compare with the face descriptors
that you get from your pipeline?  I did not do an exhaustive comparison
with your pipeline and mine, but I did on a few and the Eucledian distance
was around 0.008.  Another thing I don't understand is that your mmod face
detector provides a confidence level that is higher than 1.  I don't know
how to decipher that.
